py_zipkin/logging_helper.py

```diff
@@ -365,12 +370,58 @@ def add_span(
             duration_s,
         )
 
-        self.queue.append(thrift_span)
-        if len(self.queue) >= self.max_portion_size:
+        encoded_span = thrift.span_to_bytes(thrift_span)
+
+        # If we've already reached the max batch size or the new span doesn't
+        # fit in max_payload_bytes, send what we've collected until now and
+        # start a new batch.
+        if ((self.max_payload_bytes and
+             self.current_size + len(encoded_span) > self.max_payload_bytes) or
+                len(self.queue) >= self.max_portion_size):
```

Comment:
this formatting is a little hard to read. maybe split this out into independent statements like (use whatever variable names you like):

```python
is_within_size_limit = (self.current_size + len(encoded_span)) > (self.max_payload_bytes or -1)
is_within_portion_limit = len(self.queue) >= self.max_portion_size
if (is_within_size_limit and is_within_portion_limit):
...
```